src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp

```diff
@@ -202,13 +202,10 @@ static ColumnWithTypeAndName readColumnWithBigNumberFromBinaryData(std::shared_p
 
         for (size_t i = 0; i != chunk_length; ++i)
         {
+            /// If at least one value size is not equal to the size if big integer, fallback to reading String column and further cast to result type.
             if (!chunk.IsNull(i) && chunk.value_length(i) != sizeof(ValueType))
-                throw Exception(
-                    ErrorCodes::BAD_ARGUMENTS,
-                    "Cannot insert data into {} column from binary value, expected data with size {}, got {}",
-                    column_type->getName(),
-                    sizeof(ValueType),
-                    chunk.value_length(i));
+                return readColumnWithStringData<arrow::BinaryArray>(arrow_column, column_name);
```

Comment:
Just in case to be sure.
What happens is it would be 'toString(42424242)'?
sizeof would be 8 and as I presume chunk.value_length(i) != sizeof(ValueType) will not pass however type is a string in your test case.

Reply:
I hold in my mind the case like this
```
clickhouse local -q "select x from 
(select toString(42424242424242424242424242424242::UInt256) as x) 
UNION ALL 
(select toString(42424242424242424242424242424242424242424242::UInt256) as x) 
format Parquet" |\
 clickhouse local --input-format=Parquet --structure='x UInt256' -q "select * from table"
```

When client write all his ints, which might be or might not be (according to the sizeof()) big int, as a strings.
And if I understand you right this is ok. 

Till all sizeof(valuse) !=  32. Sizeof(UInt256). If all values by accident are 32 bytes size we will parse them as binary representation. Like this
```
clickhouse local -q "select toString(42424242424242424242424242424242::UInt256) as x format Parquet" | clickhouse local --input-format=Parquet --structure='x UInt256' -q "select * from table"
22707864971053448441042714569797161695738549521977760418632926980540162388532
```

Reply:
May be I'm missing the things that client has to follow the rule:
"write small int as int, write only big ints as string".
If so, then there is no problem.

My case applies only to the case when client just writes all ints. which might or not be big as strings.

Reply:
> May be I'm missing the point that client has to follow the rule:
"write small int as int, write only big ints as string".

I don't understand it. We are talking about only big integers, when client specified column type `Int128/UInt128/Int256/UInt256`. If client specified other integer like `Int32/UInt64...` there will be no problem at all.

So, the problem can be only when client have String column in Parquet file, specified column type `Int128/UInt128/Int256/UInt256` and column contains all values with size `sizeof(Int128/UInt128/Int256/UInt256)`. And it shouldn't be the real problem

Reply:
I'm OK with this changes.

Basically we are tuning here some specific illformed convertation. 
`select toString(42::UInt256) as x format Parquet`
That data has to be read as String.
The case when user tries to read it as UInt256 is just wrong. Here we are trying to eliminate this error by guessing a cast function. In general we has to follow with provided read schema. And only when we notice that the schema is wrong we try to guess.